firmware/libraries/versavis/src/VN100.cpp

```diff
@@ -0,0 +1,191 @@
+////////////////////////////////////////////////////////////////////////////////
+//  May 2015
+//  Author: light Dynamics and Control Lab
+//  Adapted by Florian Tschopp <ftschopp@ethz.ch> for use in VersaVIS
+////////////////////////////////////////////////////////////////////////////////
+//  VN100.cpp
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) 2018 Flight Dynamics and Control Lab
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// NOTE: Before running this code, you must configure the IMU as
+// defined in the README file.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "VN100.h"
+#include "helper.h"
+#include "versavis_configuration.h"
+
+VN100::VN100(ros::NodeHandle *nh, const String &topic, const int rate_hz,
+             Timer &timer)
+    : Imu(nh, topic, rate_hz, timer), kImuSyncTimeoutUs(4000) {
+  imu_accelerator_sensitivity_ = 1.0 / (0.00025 * 9.81);
+  imu_gyro_sensitivity_ = 1.0 / (0.05 * M_PI / 180);
+}
+
+void VN100::setup() {
+  DEBUG_PRINTLN((topic_ + " (VN100.cpp): Setup.").c_str());
+  // Start Serial1 for IMU communication
+  Serial1.begin(921600, SERIAL_8N1);
```

Comment:
I would make `Serial1` a member variable such that the user can pick between `Serial1` and `Serial`. See https://github.com/ethz-asl/versavis/pull/4/files#diff-6b7602a89dc20a4bc205c454212d864fR35

Reply:
Good idea. I added it but not as generic as you suggested because it is only valid for this single IMU and not all other supported ADIS ones.